client/src/helpers/fhir/__tests/ServiceRequestNotes.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { getNoteComment, getNoteStatus, updateNoteComment, updateNoteStatus } from '../fhir'
+import * as mocks from './ServiceRequestNotes.mocks'
+
+describe('ServiceRequest GET Note Comment', () => {
+
+  test('getNoteComment - should return note comment', () => {
+    expect(getNoteComment(mocks.serviceRequestCommentOnly)).toEqual('foo')
+  })
+
+  test('getNoteComment - should ignore -- string', () => {
+    expect(getNoteComment(mocks.serviceRequestEmptyComment)).toBeUndefined
+  })
+
+  test('getNoteComment - should be robust to missing data', () => {
+    expect(getNoteComment(null)).toBeUndefined
+    expect(getNoteComment([])).toBeUndefined
+  })
+
+})
+
+describe('ServiceRequest GET Note Status', () => {
+
+  test('getNoteStatus - should return note status', () => {
+    expect(getNoteStatus(mocks.serviceRequestCommentAndStatus)).toEqual('bar')
+  })
+
+  test('getNoteStatus - should ignore -- string', () => {
+    expect(getNoteComment(mocks.serviceRequestEmptyCommentAndStatus)).toBeUndefined
+  })
+
+  test('getNoteStatus - should be robust to missing data', () => {
+    expect(getNoteStatus(mocks.serviceRequestCommentOnly)).toBeUndefined
+    expect(getNoteStatus(null)).toBeUndefined
+    expect(getNoteStatus([])).toBeUndefined
+  })
+
+})
+
+describe('ServiceRequest UPDATE Note Comment', () => {
+
+  test('updateNoteComment - should add note comment', () => {
+    const notes = []
+    updateNoteComment(notes, {text: 'new comment'})
+    expect(notes[0].text).toEqual('new comment')
+  })
+
+  test('updateNoteComment - should replace note comment and keep previous status', () => {
+    const notes = [...mocks.notesWithCommentAndStatus]
+    updateNoteComment(notes, { text: 'new comment' })
+    expect(notes[0].text).toEqual('new comment')
+    expect(notes[1].text).toEqual('existing status')
+  })
+
+  test('updateNoteComment - should be robust to missing data', () => {
```

Comment:
I do not get this test. I think it's much better to do like you did above (around line 32). Not throwing, in my opinion, is not necessarily a sign of robustness. Plus, you don't even check for a specific error. If you disagree, than you could use jest built-in mechanism for this such as:

`expect(...).not.toThrow(myError)`

Reply:
Will be changed when updateNoteComment will return the updated notes.